src/cascade/input_data/db/asdr.py

```diff
@@ -0,0 +1,153 @@
+"""Upload asdr data to t3 table so EpiViz can access it for plotting."""
+
+import logging
+
+import pandas as pd
+
+try:
+    from db_queries import get_demographics
+except ImportError:
+
+    class DummyGetDemographics:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_queries not found")
+
+    get_demographics = DummyGetDemographics()
+
+try:
+    from db_queries import get_envelope
+except ImportError:
+
+    class DummyGetEnvelope:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_queries not found")
+
+    get_envelope = DummyGetEnvelope()
+
+
+from cascade.core.db import cursor
+from cascade.input_data.db import GBD_ROUND_ID
+
+
+CODELOG = logging.getLogger(__name__)
+
+
+def _get_age_group_data(execution_context):
+    """t3_model_version_asdr table needs age_lower and age_upper columns"""
+
+    query = """
+    SELECT age_group_id, age_group_years_start, age_group_years_end
+    FROM shared.age_group
+
+    """
+
+    with cursor(execution_context) as c:
+        c.execute(query)
+        age_group_data = pd.DataFrame(list(c.fetchall()))
+        age_group_data.columns = ["age_group_id", "age_lower", "age_upper"]
+
+    return age_group_data
+
+
+def _asdr_in_t3(execution_context):
+    """Checks if data for the current model_version_id already exists in tier 3.
+    """
+
+    model_version_id = execution_context.parameters.model_version_id
+
+    query = """
+    SELECT exists(
+             SELECT * FROM epi.t3_model_version_asdr
+             WHERE model_version_id = %(model_version_id)s
+    )
+    """
+    with cursor(execution_context) as c:
+        c.execute(query, args={"model_version_id": model_version_id})
+        exists = c.fetchone()[0]
+
+    return exists == 1
+
+
+def _get_asdr_data(execution_context):
+
+    parent_loc = execution_context.parameters.location_id
+
+    demo_dict = get_demographics(gbd_team="epi", gbd_round_id=GBD_ROUND_ID)
+    age_group_ids = demo_dict["age_group_id"]
+    sex_ids = demo_dict["sex_id"]
+
+    asdr = get_envelope(
+        location_id=parent_loc,
+        gbd_round_id=GBD_ROUND_ID,
+        age_group_id=age_group_ids,
+        sex_id=sex_ids,
+        with_hiv=1,
```

Comment:
I know that their docs and code both treat `with_hiv` and `rates` as integers but that's absurd, let's not leak it into our code. Booleans will also work and we should use those.